src/NServiceBus.Metrics/QueueLength/QueueLengthMarticBuilder.cs

```diff
@@ -0,0 +1,157 @@
+﻿namespace NServiceBus.Metrics.QueueLength
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Features;
+    using global::Metrics;
+    using global::Metrics.Core;
+    using Pipeline;
+    using Routing;
+
+    class QueueLengthMetricBuilder : MetricBuilder
+    {
+        const string HeaderName = "NServiceBus.Metrics.QueueLength";
+
+        const char SeparatorValue = '_';
+        static readonly string Separator = SeparatorValue.ToString();
+        static readonly char[] Separators = { SeparatorValue };
+
+        MetricsContext metricsContext;
+
+        ConcurrentDictionary<string, Tuple<Guid, CounterImplementation>> sendingCounters = new ConcurrentDictionary<string, Tuple<Guid, CounterImplementation>>();
+        ConcurrentDictionary<Guid, SequenceReporter> receivingReporters = new ConcurrentDictionary<Guid, SequenceReporter>();
+        static readonly Unit Unit = Unit.Custom("Sequence");
+
+        public override void Define(MetricsContext metricsContext)
```

Comment:
@danielmarbach I think there is no benefit of having this counter in the perf counters  as the raw value would not tell you anything. Only the processed value (queue length) has a meaning. 

Also, the metrics are defined dynamically based on destinations of sent messages so we can't use attributes.